src/Util/Strings.php

```diff
@@ -0,0 +1,408 @@
+    {
+        if ($network != "") {
+            if ($url != "") {
+                $network_name = '<a href="'.$url.'">'.ContactSelector::networkToName($network, $url)."</a>";
+            } else {
+                $network_name = ContactSelector::networkToName($network);
+            }
+
+            return $network_name;
+        }
+    }
+
+    /**
+     * @brief Remove intentation from a text
+     * 
+     * @param string $text  String to be transformed.
+     * @param string $chr   Optional. Indentation tag. Default tab (\t).
+     * @param int    $count Optional. Default null.
+     * 
+     * @return string       Transformed string.
+     */
+    public static function deindent($text, $chr = "[\t ]", $count = NULL)
+    {
+        $lines = explode("\n", $text);
+
+        if (is_null($count)) {
+            $m = [];
+            $k = 0;
+            while ($k < count($lines) && strlen($lines[$k]) == 0) {
+                $k++;
+            }
+            preg_match("|^" . $chr . "*|", $lines[$k], $m);
+            $count = strlen($m[0]);
+        }
+
+        for ($k = 0; $k < count($lines); $k++) {
+            $lines[$k] = preg_replace("|^" . $chr . "{" . $count . "}|", "", $lines[$k]);
+        }
+
+        return implode("\n", $lines);
+    }
+
+    /**
+     * @brief Get byte size returned in a Data Measurement (KB, MB, GB)
+     * 
+     * @param int $bytes    The number of bytes to be measured
+     * @param int $precision    Optional. Default 2.
+     * 
+     * @return string   Size with measured units.
+     */
+    public static function formatBytes($bytes, $precision = 2)
+    {
+        $units = ['B', 'KB', 'MB', 'GB', 'TB'];
+        $bytes = max($bytes, 0);
+        $pow = floor(($bytes ? log($bytes) : 0) / log(1024));
+        $pow = min($pow, count($units) - 1);
+        $bytes /= pow(1024, $pow);
+
+        return round($bytes, $precision) . ' ' . $units[$pow];
+    }
+
+    /**
+     * @brief Protect percent characters in sprintf calls
+     * 
+     * @param string $s String to transform.
+     * 
+     * @return string   Transformed string.
+     */
+    public static function protectSprintf($s)
+    {
+        return str_replace('%', '%%', $s);
+    }
+
+    /**
+     * @brief Base64 Encode URL and translate +/ to -_ Optionally strip padding.
+     * 
+     * @param string $s                 URL to encode
+     * @param boolean $strip_padding    Optional. Default false
+     * 
+     * @return string   Encoded URL
+     */
+    public static function base64UrlEncode($s, $strip_padding = false)
+    {
+        $s = strtr(base64_encode($s), '+/', '-_');
+
+        if ($strip_padding) {
+            $s = str_replace('=', '', $s);
+        }
+
+        return $s;
+    }
+
+    /**
+     * @brief Decode Base64 Encoded URL and translate -_ to +/
+     * @param string $s URL to decode
+     * 
+     * @return string   Decoded URL
+     */
+    public static function base64UrlDecode($s)
```

Comment:
Maybe we move it to  `Friendica\Util\URL` class with name `base64Decode($url)`.
Sounds more right.